operatorcourier/nest.py

```diff
@@ -2,34 +2,88 @@
 import logging
 import os
 import yaml
-import operatorcourier.identify as identify
+from shutil import copyfile
+from tempfile import TemporaryDirectory
+from operatorcourier import identify
+from operatorcourier.errors import OpCourierBadBundle
+from operatorcourier.manifest_parser import \
+    is_manifest_folder, get_csvs_pkg_info_from_root, get_crd_csv_files_info, \
+    CRD_STR, CSV_STR, PKG_STR
+
 
 logger = logging.getLogger(__name__)
 
 
-def nest_bundles(yaml_files, registry_dir, temp_registry_dir):
+def nest_bundles(source_dir, output_dir):
+    root_path, dir_names, root_dir_files = next(os.walk(source_dir))
+    csvs_info, pkg_info = get_csvs_pkg_info_from_root(source_dir)
+
+    dir_paths = [os.path.join(source_dir, dir_name) for dir_name in dir_names]
+    manifest_paths = list(filter(lambda x: is_manifest_folder(x), dir_paths))
+
+    # nested layout
+    if manifest_paths:
+        logger.warning('The source directory is already nested.')
+
+        # extract paths of package file in root dir and
+        # valid CRD/CSV files from subdirectories, and ignore irrelevant ones
+        manifest_files_path = [pkg_info[0]]
+
+        for manifest_path in manifest_paths:
+            crds_info, csvs_info = get_crd_csv_files_info(manifest_path)
+            crd_csv_file_paths = [file_info[0] for file_info in (crds_info + csvs_info)]
+            manifest_files_path.extend(crd_csv_file_paths)
+
+        # copy all manifest files to output_dir with folder structure preserved
+        source_dir = os.path.join(source_dir, '')  # adds a trailing slash if missing
+        os.makedirs(output_dir, exist_ok=True)
+        for file_path in manifest_files_path:
+            file_path_relative = file_path[len(source_dir):]
```

Comment:
Could you try this? https://docs.python.org/3/library/os.path.html#os.path.relpath

You don't need this hack with that `source_dir = os.path.join(source_dir, '')  # adds a trailing slash if missing`